src/core/Events.js

```diff
@@ -216,24 +212,55 @@ export var Events = {
 	},
 
 	// @method listens(type: String, propagate?: Boolean): Boolean
+	// @method listens(type: String, fn: Function, context?: Object, propagate?: Boolean): Boolean
 	// Returns `true` if a particular event type has any listeners attached to it.
 	// The verification can optionally be propagated, it will return `true` if parents have the listener attached to it.
-	listens: function (type, propagate) {
+	listens: function (type, fn, context, propagate) {
 		if (typeof type !== 'string') {
 			console.warn('"string" type argument expected');
 		}
+
+		if (typeof fn === 'boolean') {
+			propagate = fn;
+			fn = undefined;
+			context = undefined;
+		}
+
 		var listeners = this._events && this._events[type];
-		if (listeners && listeners.length) { return true; }
+		if (listeners && listeners.length) {
+			if (fn) {
+				if (this._listens(type, fn, context)) {
+					return true;
+				}
+			} else {
+				return true;
+			}
+		}
 
 		if (propagate) {
 			// also check parents for listeners if event propagates
 			for (var id in this._eventParents) {
-				if (this._eventParents[id].listens(type, propagate)) { return true; }
+				if (this._eventParents[id].listens(type, fn, context, propagate)) { return true; }
 			}
 		}
 		return false;
 	},
 
+	_listens: function (type, fn, context) {
+		var listeners = this._events && this._events[type] || [];

```

Comment:
I'm wondering... What if we handle the `(type)` case here by adding `if (!fn) return true`? Then we would be able to remove a block of code above (229–238) and replace it with just `if (this._listens(type, fn, context)) { return true; }`.

Reply:
Good point, changed